### ISSUE TITLE
Image Classification Pipeline returns score= 1.0

### ISSUE THREAD

**Issue by guillaumeguy** [huggingface/transformers#21303]
### System Info

Vision Transformers Documentation mentions that they do support regression when num_labels == 1. However, it seems incompatible with Pipeline.

In this code, the logits are normalized into scores. However, when num_labels = 1, it effectively turns the score to `1`. 

https://github.com/huggingface/transformers/blob/b2c863a3196150850d17548f25ee0575bccb8224/src/transformers/pipelines/image_classification.py#L112

### Who can help?

@amyeroberts @nielsr

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

1. Train a ViT on a regression  (num_labels = 1)
2. Use pipeline

![image](https://user-images.githubusercontent.com/5290678/214615264-13b5e355-a30b-4f6a-9360-41341054229a.png)

### Expected behavior

The model should returns predictions

**Comment by sgugger** [comment]
There is no pipeline available for regression tasks, you need to use the model directly and takes its outputs.

**Comment by guillaumeguy** [comment]
Thanks @sgugger! Super fast answer! 

As I found the pipelines to be very helpful. If folks want to still use them, one can just rewrite the function in the `postprocess` [function](https://github.com/huggingface/transformers/blob/main/src/transformers/pipelines/image_classification.py#L116):

```
def postprocess(self, model_outputs, top_k=5):
        if top_k > self.model.config.num_labels:
            top_k = self.model.config.num_labels

        if self.framework == "pt":
            pred = model_outputs.logits
        else:
            raise ValueError(f"Unsupported framework: {self.framework}")

        scores = pred.tolist()
        return scores
```

You can then instantiate the pipeline of the overwritten class: 
`pipe = ImageClassificationPipeline(model=model,feature_extractor=extractor,device='cuda:0')`

And run your inference:
```
def data():
    for path in paths:
        yield PILImage.open(path)


from tqdm import tqdm
scores = []
for out in tqdm(pipe(data())):
    scores.append(out)
```

**Comment by NielsRogge** [comment]
Yes that's why the pipeline is called classification, rather than regression. We would need an `ImageRegressionPipeline` for this use case ;)

**Comment by NielsRogge** [comment]
Closing this issue as it seems resolved.
